test/infrastructure/docker/controllers/dockermachine_controller.go

```diff
@@ -209,34 +232,48 @@ func (r *DockerMachineReconciler) reconcileNormal(ctx context.Context, machine *
 		dockerMachine.Status.LoadBalancerConfigured = true
 	}
 
+	// At, this stage, we are ready for bootstrap. However, if the BootstrapExecSucceededCondition is missing we add it and we
+	// requeue so the user can se the change of state before the bootstrap actually start.
```

Comment:
You are right, the main reason for this is that it makes a nicer UX when working with CAPD, because this is where the docker-machine provisioning spends most of the time.